features/expectations/body_json_example.feature

```diff
@@ -74,6 +74,7 @@ Feature: Body - JSON example
     }
     """
     Then Gavel will NOT set any errors for "body"
+    And Gavel will set "isValid" to "true" for "body"
```

Comment:
I think this is redundant to `And Request or Response is valid` in this feature. In this particular feature `And Request or Response is valid` is the intent and `isValid` set to `true` is an implementation detail, which only pollutes the behavior description.

We can test `isValid` in gavel-spec, but IMHO it should be a separate feature. Or, better, it should be tested in the "api" features, where we test exactly what Gavel returns in case of valid/invalid.

Reply:
This comment applies also to a similar change made in the other files, such as body_json_schema.feature etc.

Reply:
Agree. I will leave out implementation details from behavior suites and ensure proper assertion on the gavel.js side itself +1

Reply:
What do you think about the next definition?

```
And "body" field is valid
And "body" field is NOT valid
```

Reply:
Hmm. Previously there was no such state, so "no errors set" was equivalent to being valid. Now there are two things which represent it - (no) errors and isValid. My hunch is that the original text is perhaps too attached to the implementation by the wording which mentions "errors", while the intention only was to check whether Gavel considers that particular scenario valid or not.

Thus my proposal would be to use your formulation instead of both lines:

```diff
- Then Gavel will NOT set any errors for "body"
- And Gavel will set "isValid" to "true" for "body"
+ Then Gavel considers the field "body" to be valid
```

...which then translates in the steps implementation as something like (pseudocode):

```js
Then('Gavel considers the field "body" to be valid', () => {
  assert.isTrue(bodyIsValid);
  assert.lengthOf(bodyErrors, 0);
});
```

What do you think about this?